### PR TITLE
Added consent records to patient consultation

### LINKED ISSUE #2004: Creating a super user without creating a district results in an error
**Describe the bug**
Setup project, create a super user. It results in an error that wont resolve until there is a district record.

**Expected behavior**
Either no district should be assigned by default to the super user or a warning should appear telling the dev to load dummy data first

**Screenshots**
image

### LINKED ISSUE #7416: Feature to upload consent file per encounter
**Is your feature request related to a problem? Please describe.**
Hospitals must maintain records of the consent received from patients. Such consent must be taken on paper and uploded to CARE.

**Describe the solution you'd like**
Add a section in the consultation form, below the treatment plan called "Consent records"

- [ ] Add field for "Consent Type"- with dropdown: 
1- Consent for admission
2- Patient Code status
3- Consent for procedure
4- High risk consent
5- Others

Only when "Patient code status" is selected, an additional field for "Patient code status" with dropdown options as below appear. 
Do Not Hospitalise (DNH)
Do Not Resuscitate (DNR)
Comfort Care Only
Active treatment (Default)

The user must be able to select an option from the Consent Type dropdown and upload a file and add a text title to the file (the existing file upload component may be used).

For the same consent type, multiple files may be uploaded.

Below this, a list of already uploaded files, arranged in sections for each consent type must be available.

- [ ] On the patient dashboard, (https://care.coronasafe.in/facility/657c32be-d584-476c-9ce2-0412f0e7692e/patient/ba1e2de6-6ded-456c-836a-de25fcc1f4c7/consultation/c50a8ccf-ee9b-42ec-beb8-b5f62e91cb12) add a section at the bottom of "Overview" for "Consent Record" and show the uploaded consent records. 

NOTE: the minimum urgent requirement for now is to have the ability to upload "Consent for admission" alone. That may be taken up first and then improved with other types of consent later.

### PR DESCRIPTION
**Pull request by shivankacker**
## Proposed Changes

- Adds consent records to patient consultation
- Fixes a bug with creating super user without injesting data.
- Adds new file upload type

### Associated Issue

Fixes https://github.com/coronasafe/care_fe/issues/7416
Fixes #2004 

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/api/serializers/file_upload.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ def check_permissions(file_type, associating_id, user, action="create"):
             return associating_id
         elif file_type == FileUpload.FileType.COMMUNICATION.value:
             return associating_id
+        elif file_type == FileUpload.FileType.CONSENT_RECORD.value:
```
Comment (sainak) [inline care/facility/api/serializers/file_upload.py]: use this block instead, as access checks are needed
https://github.com/coronasafe/care/blob/ce7073e4d789dd19272d257cc79c2500d3d53e75/care/facility/api/serializers/file_upload.py#L35-L36

File: care/facility/models/patient_consultation.py [hunk 2]
```diff
@@ -238,6 +238,8 @@ class PatientConsultation(PatientBaseModel, ConsultationRelatedPermissionMixin):
     prn_prescription = JSONField(default=dict)
     discharge_advice = JSONField(default=dict)
 
+    consent_records = JSONField(default=list)
```
Comment (sainak) [inline care/facility/models/patient_consultation.py]: I think having a fk to the consent record file upload would be better

**Review by sainak — CHANGES_REQUESTED** [review]
changes requested

File: care/facility/models/patient_consultation.py (hunk 2 above)
Comment (shivankacker) [reply care/facility/models/patient_consultation.py]: there are multiple types of consent records, and multiple files can be uploaded for each. This one just keeps track of which types have been selected. I did not add an FK as the file upload is already linked to the consultation

File: care/facility/api/serializers/file_upload.py (hunk 1 above)
Comment (shivankacker) [reply care/facility/api/serializers/file_upload.py]: made changes

**Comment by shivankacker** [comment]
@sainak can you review

File: care/facility/models/patient_consultation.py (hunk 2 above)
Comment (sainak) [reply care/facility/models/patient_consultation.py]: okay

**Review by sainak — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
image

the PR is not working , unable to upload the consent file

CC: @rithviknishad @Ashesh3 @sainak

**Comment by nihal467** [comment]
@skks1212 
image

the error still exists when we try to fill it while we are creating a new consultation

**Comment by sainak** [comment]
@skks1212 tests are required
and delete the current migrations and rerun makemigrations

**Review by sainak — APPROVED** [review]

File: care/facility/api/serializers/file_upload.py [hunk 3]
```diff
@@ -32,8 +32,18 @@ def check_permissions(file_type, associating_id, user, action="create"):
             if not has_facility_permission(user, patient.facility):
                 raise Exception("No Permission")
             return patient.id
-        elif file_type == FileUpload.FileType.CONSULTATION.value:
-            consultation = PatientConsultation.objects.get(external_id=associating_id)
+        elif (
+            file_type == FileUpload.FileType.CONSULTATION.value
+            or file_type == FileUpload.FileType.CONSENT_RECORD.value
```
Comment (vigneshhari) [inline care/facility/api/serializers/file_upload.py]: Why are we checking with or when we recheck the same condition again inside ?

**Comment by nihal467** [comment]
LGTM

File: care/facility/models/patient_consultation.py [hunk 4]
```diff
@@ -289,6 +296,18 @@ def save(self, *args, **kwargs):
...
         """
+
+        deleted_consent_records = filter(
+            lambda record: record.get("deleted", False) is True, self.consent_records
+        )
+        deleted_consent_record_ids = [
+            record.get("id") for record in deleted_consent_records
+        ]
+
+        FileUpload.objects.filter(associating_id__in=deleted_consent_record_ids).update(
```
Comment (vigneshhari) [inline care/facility/models/patient_consultation.py]: Authorization issue found, needs to be redesigned. cc @sainak

**Review by vigneshhari — CHANGES_REQUESTED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**